packages/beacon-node/src/network/processor/gossipHandlers.ts

```diff
@@ -118,6 +139,57 @@ export function getGossipHandlers(modules: ValidatorFnsModules, options: GossipH
     }
   }
 
+  async function validateBeaconBlob(
+    signedBlob: deneb.SignedBlobSidecar,
+    blobBytes: Uint8Array,
+    gossipIndex: number,
+    peerIdStr: string,
+    seenTimestampSec: number
+  ): Promise<BlockInput | null> {
+    const slot = signedBlob.message.slot;
+    const blockHex = prettyBytes(signedBlob.message.blockRoot);
+    const delaySec = chain.clock.secFromSlot(slot, seenTimestampSec);
+    const recvToVal = Date.now() / 1000 - seenTimestampSec;
+
+    const {blockInput, blockInputMeta} = getBlockInput.getGossipBlockInput(config, {
```

Comment:
if block hasn't come I think we should search for unknown block root here `network.searchUnknownSlotRoot({slot, root: rootHex});`

Reply:
there is no gurrantee which comes first, block or blobs, unkown parent search will be triggered anyway if the next slot's block or blob comes (i will confirm the same)

Reply:
may be we start search when delay > some 4000 ms?

Reply:
have added a todo for this optimization to be done in the followups